signals/ParseSignals.sed

```diff
@@ -0,0 +1,99 @@
+# htop - signals/ParseSignals.sed
+# (C) 2023 htop dev team
+# Released under the GNU GPLv2+, see the COPYING file
+# in the source distribution for its full text.
+
+# Parse a preprocessed C file with a list of signal numbers and names; output
+# the numbers and names in a sortable format.
+
+# The reason why this script exists at all is that the preprocessed C output
+# doesn't correspond line-to-line with the original source file. Among other
+# issues, it contains preprocessor directives with line numbers, which break
+# up the output. To allow sorting the file by signal name at compile time,
+# the preprocessed output has to be "parsed" and reformatted.
+
+# The script should be relatively robust wrt. C compiler output, but it should
+# still be considered a hack. It is possible that unforeseen input breaks it.
+
+# It works by building up each multiline signal entry (number and name) in the
+# hold space until it is complete, postprocessing and printing it. The entry
+# is recognized by the htop_sig{start,mid,end} markers, because these are all
+# but guaranteed not to be mangled or duplicated by the preprocessing, unlike
+# e.g. parens, braces or semicolons.
+
+
+# Skip preprocessor directives. This breaks the relationship between input and
+# output files, but it makes sorting easy.
+/^#/ d
```

Comment:
```suggestion
/^[[:space:]]*#/ d
/^[[:space:]]*%:/ d
```

Reply:
Nice find, thank you. I will also add coverage of the trigraph `??=` in addition to the digraph to the next version of the pull request, after we resolve the rest of the issues.